Library/ViewModels/LoginToutViewModel.swift

```diff
@@ -1,12 +1,29 @@
+import AuthenticationServices
 import FBSDKLoginKit
 import KsApi
 import Prelude
 import ReactiveSwift
 
+public typealias SignInWithAppleData = (appId: String, firstName: String?, lastName: String?, token: String)
+
 public protocol LoginToutViewModelInputs {
+  /// Call when Continue withApple button is pressed
+  func appleLoginButtonPressed()
+
   /// Call to set the reason the user is attempting to log in
   func configureWith(_ intent: LoginIntent, project: Project?, reward: Reward?)
 
+  /// Call when Apple completes authorization
+  @available(iOS 13.0, *)
+  func appleAuthorizationDidComplete(with data: SignInWithAppleData?)
+
+  /// Call when Apple completes authorization with error
+  @available(iOS 13.0, *)
+  func appleAuthorizationDidComplete(with error: Error)
+
+  /// Call when the SignInWithAppleEnvelope is received from the server
+  func didReceiveSignInWithAppleEnvelope(_ envelope: SignInWithAppleEnvelope)
```

Comment:
I think usually we try to name our outputs in a way that describes the action that will take place by calling that output, for example `postNotification`, `dismissKeyboard`, etc. From what I understand, what will happen when the Apple envelope is received is we will attempt to fetch the v1 `User` object to log the user into our `AppEnvironment`. I wonder if then, this output should be `loginToEnvironment`?

Reply:
Whoops this is not an output. I meant to write it next to the `didSignInWithApple` output.

Reply:
There's a difference here though. The `didSignInWithApple` emits when AuthorizationServices grants us an authorization, but we will only login to environment once we fetch the user from v1. We also already have an output `logIntoEnvironment` that's used after the facebook login flow is completed. We will use this same output to login with Apple.

Reply:
Yeah, I guess I'm wondering why we need an additional output then? Can't we just fetch the user immediately in the VM when the `signInWithApple` call succeeds, and then call `logIntoEnvironment` when the `fetchUser` call succeeds?

Reply:
The main motivation was to test the values of the `appleSignInEvent`. Once we implement the  logic to fetch user, maybe this won't be necessary anymore, because we can simply test if `logIntoEnvironment` emits after signing in with Apple

Reply:
Ended up deleting it 👍